v19.1/performance-best-practices-overview.md

```diff
@@ -57,24 +57,121 @@ This happens because when data is interleaved, queries that work on the parent t
+The approaches described above are likely to create hotspots for both reads and writes in CockroachDB. To avoid this issue, we recommend the following approaches (listed in order from best to worst performance). Each approach has pros and cons, however:
 
-The bottleneck with the second approach is that IDs generated temporally near each other have similar values and are located physically near each other in a table. This can cause a hotspot for reads and writes in a table.
+| Approach                                                                             | Pros                                             | Cons                                                                                    |
+|--------------------------------------------------------------------------------------+--------------------------------------------------+-----------------------------------------------------------------------------------------|
+| 1. [Use multi-column primary keys](#use-multi-column-primary-keys)                   | Potentially fastest, if done right               | Complex, requires up-front design and testing to ensure performance                     |
+| 2. [Use `UUID` to generate unique IDs](#use-uuid-to-generate-unique-ids)             | Good performance; spreads load well; easy choice | May leave some performance on the table; requires other columns to be useful in queries |
+| 3. [Use `INSERT` with the `RETURNING` clause](#use-insert-with-the-returning-clause-to-generate-unique-ids) | Easy to query against; familiar design           | Slower performance than the other options; higher chance of transaction contention      |
 
-The best practice in CockroachDB is to generate unique IDs using the `UUID` type, which generates random unique IDs in parallel, thus improving performance.
+### Use multi-column primary keys
+
+A well-designed multi-column primary key can yield even better performance than a [UUID primary key](#use-uuid-to-generate-unique-ids), but it requires more up-front schema design work. To get the best performance, ensure that any monotonically increasing field is located **after** the first column of the primary key. When done right, such a composite primary key should result in:
+
+- Enough randomness in your primary key to spread the table data / query load relatively evenly across the cluster, which will avoid hotspots.  By "enough randomness" we mean that the prefix of the primary key should be relatively uniformly distributed over its domain, whatever that is (and its domain should have at least as many elements as you have nodes).
+- A monotonically increasing column that is part of the primary key (and thus indexed) which is also useful in your queries.
+
+For example, consider a social media website. Social media posts are written by users, and on login the user's last 10 posts are displayed. A good choice for a primary key might be `(username, post_timestamp)`. For example:
```

Comment:
We are trying to unify the dataset (or just the story/context) around most of our SQL examples. We've chosen MovR as the dataset of choice for that initiative. I think we should try to stick to using MovR or "MovR-compliant" SQL examples for all new examples.

A separate conversation:
@thatnerd @jseldess What do you think about unifying the dataset that we use in the training documentation? Would MovR fit in that context as well?

Reply:
I agree - however this PR has been in progress for long enough that I really need to wrap it up and move on to other work.  I think updating to use a movr-ish example should probably happen in a follow-up PR at this point.